engine-tests/src/test/java/org/terasology/documentation/ModuleMarkupScraper.java

```diff
@@ -48,69 +50,85 @@
-            Module latestVersion = moduleManager.getRegistry().getLatestModuleVersion(moduleId);
-            if (!latestVersion.isOnClasspath()) {
-                gameModules.add(latestVersion);
-                System.out.println("Found Game Module: " + latestVersion.getId());
-            }
-        }
+        String wiki = generateWiki(moduleManager, sortedGameModules, sortedRemoteModules);
 
-        List<Module> remoteModules = Lists.newArrayList();
-        Iterable<Module> remoteModuleRegistry = moduleManager.getInstallManager().getRemoteRegistry();
-        Set<Name> filtered = ImmutableSet.of(TerasologyConstants.ENGINE_MODULE, new Name("engine-test"));
-        for (Module remote : remoteModuleRegistry) {
-            remoteModules.add(remote);
-            System.out.println("Found Remote Module: " + remote.getId());
-        }
-        allModules.addAll(gameModules);
-        allModules.addAll(remoteModules);
-        allModules.sort(moduleInfoComparator);
+        String fileName = "Module_Wiki.md";
+        BufferedWriter writer = new BufferedWriter(new FileWriter(new File(fileName)));
+        writer.write(wiki);
+        writer.flush();
+        writer.close();
+        System.out.println("Module Wiki generation file is ready!");
 
-        List<Module> allSortedModules;
-        allSortedModules = new ArrayList<>(allModules);
+        env.close();
+    }
 
+    private static String generateWiki(ModuleManager moduleManager, List<Module> sortedGameModules, List<Module> sortedRemoteModules) {
         StringBuilder wiki = new StringBuilder();
 
+        List<Module> allModules = Lists.newArrayList();
+        allModules.addAll(sortedGameModules);
+        allModules.addAll(sortedRemoteModules);
+        allModules.sort(MODULE_INFO_COMPARATOR);
+
+        wiki.append(generateModuleTableOfContents(sortedGameModules, sortedRemoteModules));
+
+//        moduleMapListing.append("| ");
+//        for (Module gModule : sortedGameModules) {
+//            // Add anchor tags, for example [AdditionalFruits](#AdditionalFruits)
+//            moduleMapListing.append("[").append(gModule.getId()).append("](#").append(gModule.getId()).append(") ");
+//        }
+//        moduleMapListing.append(" | ");
+//        for (Module rModule : sortedRemoteModules) {
+//            moduleMapListing.append("[").append(rModule.getId()).append("](#").append(rModule.getId()).append(") ");
+//        }
+//        moduleMapListing.append(" | ");
+//
+//        String moduleMap = System.getProperty("line.separator") +
+//                "**Extensions:**" +
+//                System.getProperty("line.separator") +
+//                "| Installed | Remote |" +
+//                System.getProperty("line.separator") +
+//                "| -------- | ---------- |" +
+//                System.getProperty("line.separator") +
+//                moduleMapListing;
+//        wiki.append(moduleMap);
```

Comment:
Is this a left-over, or some planned feature not working correctly yet?

Reply:
removed, wiki should generate as expected.